### PR TITLE
Intial Commit Review.

### PR DESCRIPTION
**Pull request by KevinJoiner**
This PR allows anyone to make comments or suggestions on the initial code.

### CODE REVIEW

File: internal/codegen/codegen.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		fmt.Printf("error formatting go source: %v\n", err)
+		formatted = goData
+	}
+	goOutputFile, err := os.Create(outputPath)
+	if err != nil {
+		return fmt.Errorf("error creating output file: %w", err)
+	}
+	defer func() {
+		if cerr := goOutputFile.Close(); err == nil && cerr != nil {
+			err = cerr
```
Comment (elffjs) [inline internal/codegen/codegen.go:80]: Will this actually bubble up?

File: internal/codegen/signal.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+	DataType   string
+	Unit       string
+	Min        string
+	Max        string
+	Desc       string
+	Deprecated bool
+
+	// Derived
+	IsArray    bool
+	GOName     string
```
Comment (elffjs) [inline internal/codegen/signal.go:36]: Hm, is this the casing you want?

File: internal/codegen/codegen.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import (
+	_ "embed"
+	"fmt"
+	"os"
+
+	"golang.org/x/tools/imports"
+)
+
+const (
+	ClickhouseFileName  = "vss-table.sql"
```
Comment (elffjs) [inline internal/codegen/codegen.go:13]: Ooh you've got a strict linter.

File: internal/codegen/codegen.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
+// Package codegen provides the code generation functionality for converting VSPEC signals to Go structs and ClickHouse tables.
+package codegen
+
+import (
+	_ "embed"
```
Comment (elffjs) [inline internal/codegen/codegen.go:5]: I swear there's an exception for this one in the linter.

Comment (KevinJoiner) [reply internal/codegen/codegen.go]: I used to have the templates embedded in this file before I moved them to separate packages, so this import was an artifact.

File: schema/migrations.json [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+        "vspecName": "Vehicle.DIMO.VehicleID",
+        "conversion": {
+            "originalName": "data.vehicleID",
+            "originalType": "string"
+        }
+    },
+    {
+        "vspecName": "Vehicle.DIMO.DefinitionID",
+        "conversion": {
+            "originalName": "data.definitionID",
```
Comment (elffjs) [inline schema/migrations.json:48]: I don't recognize these fields.

**Pull request closed without merging**